### ISSUE TITLE
Fix MLflow deprecations, remove LightGBM, and clean up training pipeline

### ISSUE THREAD

**Issue by nateeatsrice** [nateeatsrice/fraud-detection#28]
## Problem

Running `python src/train.py --data data/transactions.csv` produces four categories of warnings and errors that need to be addressed before they become breaking changes or continue polluting logs.

### 1. MLflow FileStore backend is deprecated (CRITICAL)

```
FutureWarning: The filesystem tracking backend (e.g., './mlruns') will be deprecated
in February 2026. Consider transitioning to a database backend (e.g., 'sqlite:///mlflow.db')
```

The `file://` tracking URI we currently use is being removed. This will eventually **break** the training pipeline entirely — not just warn.

### 2. Corrupted `mlruns/` experiment directory

```
Malformed experiment '1'. Detailed error Yaml file
'/workspaces/fraud-detection/mlruns/1/meta.yaml' does not exist.
```

An orphaned experiment directory exists at `mlruns/1/` with no `meta.yaml`. The filesystem backend is fragile and cannot self-heal from partial writes. This error repeats on every run.

### 3. `artifact_path` positional argument deprecated in `mlflow.sklearn.log_model`

```
WARNING mlflow.models.model: `artifact_path` is deprecated. Please use `name` instead.
```

This fires on every single `log_model` call (15 times per full training run), flooding the output with noise that masks real errors.

### 4. LightGBM feature name warnings

```
UserWarning: X does not have valid feature names, but LGBMClassifier was fitted with feature names
```

LightGBM's sklearn wrapper captures feature names during `.fit()` and then warns when `.predict()` receives a bare numpy array without matching names. This is a design incompatibility between how we pass data (as `.values` numpy arrays) and how LightGBM expects it.

**Decision:** Rather than refactoring the entire data pipeline to pass DataFrames (which would touch `train.py`, `test_train.py`, `app/main.py`, and `flow.py`), we will **remove LightGBM entirely**. XGBoost provides equivalent gradient boosting capability without this issue, and removing LightGBM also reduces the Docker image size and the dependency surface.

## Acceptance criteria

- [ ] MLflow tracking URI defaults to `sqlite:///` instead of `file://`
- [ ] Old `mlruns/` directory removed; `.gitignore` updated to ignore it
- [ ] New `mlartifacts/` directory used for artifact storage, added to `.gitignore`
- [ ] `mlflow.sklearn.log_model` uses `name=` keyword argument
- [ ] LightGBM removed from: `src/train.py`, `app/main.py`, `pyproject.toml`, `requirements.txt`, `Dockerfile` comments
- [ ] All tests pass (`pytest -q`)
- [ ] Training run completes with **zero** warnings from the 4 categories above

## Files affected

| File | Change |
|---|---|
| `src/train.py` | SQLite backend, remove LightGBM, `name=` kwarg |
| `app/main.py` | Remove LightGBM from `PredictionRequest.model_name` description |
| `pyproject.toml` | Remove `lightgbm>=4.0.0` from dependencies |
| `requirements.txt` | Remove `lightgbm>=4.0.0` |
| `Dockerfile` | Update `libgomp1` comment (still needed for XGBoost) |
| `.gitignore` | Add `mlartifacts/`, keep `mlruns/` ignored |
| `README.md` | Update references from LightGBM to reflect current model list |
